lib/Dialect/HW/Transforms/LowerHWtoBTOR2.cpp

```diff
@@ -0,0 +1,786 @@
+  // associating it with the current lid
+  void setConstLID(int64_t val, size_t w) {
+    // Create a new entry for the pair
+    constToLIDMap[std::make_pair<int64_t, size_t>(
+        std::forward<int64_t>(val), std::forward<size_t>(w))] = lid;
+  }
+
+  // Checks if a sort was declared with the given width
+  // If so, its lid will be returned
+  // Otherwise -1 will be returned
+  size_t getSortLID(size_t w) {
+    // Look for presence of the width in a pair
+    for (auto p : sortToLIDMap) {
+      // If the width was declared, return it's lid
+      if (w == p.first) {
+        return p.second;
+      }
+    }
+    // If no lid was found return -1
+    return NO_LID;
+  }
+
+  // Updates or creates an entry for the given width
+  // associating it with the current lid
+  void setSortLID(size_t w) {
+    // Check for the existence of the sort
+    for (size_t i = 0; i < sortToLIDMap.size(); ++i) {
+      auto p = sortToLIDMap[i];
+
+      // If the width was declared, update it's lid
+      if (w == p.first) {
+        sortToLIDMap[i] = std::make_pair<size_t, size_t>(
+            std::forward<size_t>(w), std::forward<size_t>(lid));
+        return;
+      }
+    }
+
+    // Otherwise simply create a new entry
+    sortToLIDMap.push_back(std::make_pair<size_t, size_t>(
+        std::forward<size_t>(w), std::forward<size_t>(lid)));
+  }
+
+  // Checks if an operation was declared
+  // If so, its lid will be returned
+  // Otherwise -1 will be returned
+  size_t getOpLID(Operation *op) {
+    // Look for the operation declaration
+    Operation *defOp = getOpAlias(op);
+    if (opLIDMap.contains(defOp)) {
+      return opLIDMap[defOp];
+    }
+
+    // If no lid was found return -1
+    return NO_LID;
+  }
+
+  // Checks if an operation was declared
+  // If so, its lid will be returned
+  // Otherwise -1 will be returned
+  size_t getOpLID(Value op) {
+    // Look for the operation declaration
+    Operation *defOp = getOpAlias(op.getDefiningOp());
+
+    if (opLIDMap.contains(defOp)) {
+      return opLIDMap[defOp];
+    }
+
+    // Check for special case where op is actually a port
+    // To do so, we start by checking if our operation isa block argument
+    if (BlockArgument barg = dyn_cast<BlockArgument>(op)) {
+      // Extract the block argument index and use that to get the line number
+      size_t argIdx = barg.getArgNumber();
+
+      // Check that the extracted argument is in range
+      if (inputLIDs.contains(argIdx)) {
+        return inputLIDs[argIdx];
+      }
+    }
+
+    // If no lid was found return -1
+    return NO_LID;
+  }
+
+  // Updates or creates an entry for the given operation
+  // associating it with the current lid
+  void setOpLID(Operation *op) {
+    if (op != nullptr) {
+      opLIDMap[op] = lid;
+    }
+  }
+
+  // Checks if an operation has an alias
+  // If so, the original operation is returned
+  // Otherwise the argument is returned as it is the original op
```

Comment:
Somewhere is there something explaining what op aliases do?  What problem are they solving?